clusterloader2/pkg/measurement/common/service_creation_latency.go

```diff
@@ -230,24 +259,24 @@ func (s *serviceCreationLatencyMeasurement) deleteObject(svc *corev1.Service) er
 }
 
 func (s *serviceCreationLatencyMeasurement) updateObject(svc *corev1.Service) error {
-	// This measurement only works for services with LoadBalancer type.
-	if svc.Spec.Type != corev1.ServiceTypeLoadBalancer {
+	// This measurement only works for services with ClusterIP, NodePort and LoadBalancer type.
+	if svc.Spec.Type != corev1.ServiceTypeClusterIP && svc.Spec.Type != corev1.ServiceTypeNodePort && svc.Spec.Type != corev1.ServiceTypeLoadBalancer {
 		return nil
 	}
 	key, err := runtimeobjects.CreateMetaNamespaceKey(svc)
 	if err != nil {
 		return fmt.Errorf("meta key created error: %v", err)
 	}
-	if _, exists := s.creationTimes.Get(key, creatingPhase); !exists {
-		s.creationTimes.Set(key, creatingPhase, svc.CreationTimestamp.Time)
+	if _, exists := s.creationTimes.Get(key, phaseName(creatingPhase, svc.Spec.Type)); !exists {
+		s.creationTimes.Set(key, phaseName(creatingPhase, svc.Spec.Type), svc.CreationTimestamp.Time)
 	}
-	if len(svc.Status.LoadBalancer.Ingress) < 1 {
+	if svc.Spec.Type == corev1.ServiceTypeLoadBalancer && len(svc.Status.LoadBalancer.Ingress) < 1 {
 		return nil
 	}
-	if _, exists := s.creationTimes.Get(key, ipAssigningPhase); exists {
+	if _, exists := s.creationTimes.Get(key, phaseName(ipAssigningPhase, svc.Spec.Type)); exists {
```

Comment:
I'm wondering if those metrics even make any sense for non-loadbalancer services.
Because for ClusterIP IP is assigned as part of service creation API call.

Maybe add a comment (where the metrics are defined), that we're aware of it, just add those for simplicity of the code

Reply:
I have remove ipAssigning phase in ClusterIP and NodePort type of service.